### ISSUE TITLE
Issue with uniquifying circuits with verilog field

### ISSUE THREAD

**Issue by leonardt** [phanrahan/magma#330]
```
HalfAdder = m.DefineCircuit('HalfAdder', 
   'A', m.In(m.Bit), 
   'B', m.In(m.Bit),
   'S', m.Out(m.Bit),
   'C', m.Out(m.Bit))
HalfAdder.verilog  = '''\
    assign S = A ^ B;
    assign C = A & B;\
'''
m.EndCircuit()
```

produces the warning

```
magma:WARNING:Could not compile circuit: 'Got None for port 'HalfAdder.S', is it connected to anything?'. Uniquifying anyway.
```

I suspect it's because it's trying to compile the circuit, but it's not checking that the definition is contained with the `verilog` string rather than being a standard magma definition. The issue could be that the coreir backend is not properly handling the `verilog` field (it should skip compiling if this is the case and pass the verilog string to coreir).

**Comment by rsetaluri** [comment]
In general circuits which wrap verilog can be compiled to coreir, see https://github.com/phanrahan/magma/blob/40cf2e082c5f50f388317ab3d3fba2ba8a60ddd0/magma/backend/coreir_.py#L306-L309.

The issue here is that wrapped verilog are assumed to have the attribute `verilogFile`, where as here we set `.verilog`. I see that we set `.verilog` when compiling to verilog, so perhaps we can unify these two fields or formalize this another way, such as having a `VerilogCircuit` metaclass as well?
